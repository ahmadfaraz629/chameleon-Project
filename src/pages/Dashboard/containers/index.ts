import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import { fetchUsersAction } from 'store/actions/user';
import { getSortedUsers, getSummary } from 'store/selectors/user';
import { IState } from 'types';
import Dashboard from 'pages/Dashboard';

const mapStateToProps = (state: IState) => {
  return {
    users: getSortedUsers(state),
    summary: getSummary(state)
  };
};

const mapDispatchToProps = (dispatch: Dispatch) => {
  return bindActionCreators({ fetchUsers: fetchUsersAction.STARTED }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
