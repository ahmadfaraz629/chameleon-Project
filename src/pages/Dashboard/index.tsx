import React, { FC, useEffect } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import SummaryComponent from 'components/Summary';
import UsersComponent from 'components/User';
import { IUser, ISummary } from 'types';

interface IDashboardProps {
  fetchUsers: () => void;
  summary: ISummary;
  users: IUser[];
}

const useStyles = makeStyles(theme => ({
  container: {
    paddingTop: theme.spacing(5)
  }
}));

const Dashboard: FC<IDashboardProps> = ({ fetchUsers, summary, users }) => {
  const classes = useStyles();

  useEffect(() => {
    fetchUsers();
  }, [fetchUsers]);

  return (
    <>
      <Container className={classes.container}>
        <SummaryComponent summary={summary} />
        <UsersComponent users={users} />
      </Container>
    </>
  );
};

export default Dashboard;
