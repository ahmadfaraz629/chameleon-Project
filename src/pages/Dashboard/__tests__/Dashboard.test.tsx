// Dashboars.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import Dashboard from 'pages/Dashboard';
import { DashboardProps } from 'pages/Dashboard/fixture';

test('Dashboard Snapshot Test', () => {
  const component = renderer.create(<Dashboard {...DashboardProps}></Dashboard>);

  let dashboard = component.toJSON();
  expect(dashboard).toMatchSnapshot();
});
