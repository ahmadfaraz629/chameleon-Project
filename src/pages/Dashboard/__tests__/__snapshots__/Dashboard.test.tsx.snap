// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Snapshot Test 1`] = `
<div
  className="MuiContainer-root makeStyles-container-1 MuiContainer-maxWidthLg"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3"
    >
      <div
        className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation4 MuiPaper-rounded"
      >
        <h6
          className="MuiTypography-root makeStyles-typographyValue-3 MuiTypography-subtitle1"
        >
          <span />
        </h6>
        <p
          className="MuiTypography-root makeStyles-typographyLabel-4 MuiTypography-body1"
        >
          Team Member
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3"
    >
      <div
        className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation4 MuiPaper-rounded"
      >
        <h6
          className="MuiTypography-root makeStyles-typographyValue-3 MuiTypography-subtitle1"
        >
          -
        </h6>
        <p
          className="MuiTypography-root makeStyles-typographyLabel-4 MuiTypography-body1"
        >
          Last User added
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3"
    >
      <div
        className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation4 MuiPaper-rounded"
      >
        <h6
          className="MuiTypography-root makeStyles-typographyValue-3 MuiTypography-subtitle1"
        >
          <span />
        </h6>
        <p
          className="MuiTypography-root makeStyles-typographyLabel-4 MuiTypography-body1"
        >
          Users Invited
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-3"
    >
      <div
        className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation4 MuiPaper-rounded"
      >
        <h6
          className="MuiTypography-root makeStyles-typographyValue-3 MuiTypography-subtitle1"
        >
          <span />
        </h6>
        <p
          className="MuiTypography-root makeStyles-typographyLabel-4 MuiTypography-body1"
        >
          Published Campaigns
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root makeStyles-root-5 MuiPaper-elevation10 MuiPaper-rounded"
  >
    <div
      className="MuiTableContainer-root makeStyles-container-6"
    >
      <table
        aria-label="sticky table"
        className="MuiTable-root MuiTable-stickyHeader"
        role={null}
      >
        <thead
          className="MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
              scope="col"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
              scope="col"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Name
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
              scope="col"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Email
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
              scope="col"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Role
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
              scope="col"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              User Invited
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
              scope="col"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Published Campaigns
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
              scope="col"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Member Since
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-hover"
            role={null}
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <div
                className="MuiAvatar-root MuiAvatar-circle"
              >
                <img
                  alt="Aaron Cody"
                  className="MuiAvatar-img"
                  src="https://d1ts43dypk8bqh.cloudfront.net/v1/avatars/eb38d065-c587-402f-aa17-135c754db73a"
                />
              </div>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Aaron Cody
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              aaron@trychameleon.com
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              owner
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              3
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              0
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              17 Sep 2015
            </td>
          </tr>
          <tr
            className="MuiTableRow-root MuiTableRow-hover"
            role={null}
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <div
                className="MuiAvatar-root MuiAvatar-circle"
              >
                <img
                  alt="Andrew Hendricks"
                  className="MuiAvatar-img"
                  src="https://lh3.googleusercontent.com/-Av9tgNfU2eE/AAAAAAAAAAI/AAAAAAAAAAk/CUY-pphPtNE/s50/photo.jpg"
                />
              </div>
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Andrew Hendricks
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              andrew@trychameleon.com
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              member
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              0
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              0
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body"
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              2 May 2019
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
